DragonLens.cs

```diff
@@ -37,5 +38,43 @@ public override void HandlePacket(BinaryReader reader, int whoAmI)
 			if (type == "PlayerManager")
 				PlayerManagerNetHandler.HandlePacket(reader);
 		}
+
+
+		public override object Call(params object[] args)
+		{
+			const string Tag = "AddAPITool";
+
+			try
+			{
+				// ── Validate signature ──────────────────────────────────────────────
+				if (args is not { Length: 4 } ||
+					args[0] is not string tag ||
+					!tag.Equals(Tag, StringComparison.Ordinal))
+				{
+					throw new ArgumentException(
+						$"{Tag} Error: Expected (\"{Tag}\", string name, Asset<Texture2D> icon, Action click).");
+				}
+
+				// ── Safe‑cast with fall‑backs ───────────────────────────────────────
+				string name = args[1] as string ?? "Unnamed";
+				Asset<Texture2D> icon = args[2] as Asset<Texture2D> ?? TextureAssets.MagicPixel;
+				Action click = args[3] as Action ?? (() => Logger.Error($"{Tag} Error: Click action was null."));
+
+				// Log any fall‑backs that were actually used
+				if (args[1] is null) Logger.Error($"{Tag} Error: Name was null – defaulted to \"{name}\".");
+				if (args[2] is null) Logger.Error($"{Tag} Error: Icon was null – defaulted to TextureAssets.MagicPixel.");
+				if (args[3] is null) Logger.Error($"{Tag} Error: Click action was null – defaulted to no‑op.");
+
+				// Constructor registers itself; nothing more to do
+				return new APITool(this, name, icon, click);
+			}
+			catch (Exception ex)
+			{
+				Logger.Error($"[DragonLens Call Error] {ex}");
+				return null;
+			}
+		}
+
+
 	}
 }
```

Comment:
Some other notes and pieces of functionality

- Likely need to expose an API to interact with the style manager and retrieve colors/box rendering, and ideally the UI superclasses to allow that to be extended. Any advanced usecase will likely need to integrate with the UI and having that exposed and documented to allow them to use the styling providers is a must.

- The API tool completely lacks the functionality to override the net sync methods, thats a hard requirement and is important for anything multiplayer related.

- It may be worthwhile to add visual indication of which tools are from which mod, such as appending to their tooltip

-P.S. sorry it took me a bit to get around to reviewing this >.<